### PR TITLE
Rename style -> length for DateTimeFormat everywhere

### LINKED ISSUE #273: [datetime] Style vs Length vs Width
There's a combination of inconsistent names around what ECMA402 calls `dateStyle` and `timeStyle`. Shane raised this as a question whether we really should follow ECMA402 here and name it `style`, or use another term.

UTS #35 has a name `dateFormatLength`and `timeFormatLength` - https://unicode.org/reports/tr35/tr35-dates.html#dateFormats

```
<!ATTLIST dateFormatLength type ( full | long | medium | short ) #REQUIRED >
<!ATTLIST timeFormatLength type ( full | long | medium | short ) #REQUIRED >
```

and is followed by this sentence:

> Standard date and time patterns are each normally provided in four **types**: full (usually with weekday name), long (with wide month name), medium, and short (usually with numeric month).

(emphasis mine)

So, we could use widths, lengths, types, or styles. 

My problem with `styles` is that it misses an opportunity to convey that this is about how long the returned value will be. Style can be "Informal"/"Formal" or "Artistic"/"Technical" etc. "Width" and "Length" are scoped down to clearly indicate that we're talking about widths.

On the other hand, `StyleBag` seems reasonable, but `LengthBag` and `WidthBag` kind of feel off linguistically.
We also unlikely going to have another use for the word `style` in DateTime context, so we could let the wider semantical scope be occupied by it I think.

Either way, if we want to stick to ECMA402, we could suggest naming refactor there and see what comes out of it. For us, chosing and changing the name is not as drastic, especially early on.

### PR DESCRIPTION
**Pull request by nciric**
Fixes #273

### CODE REVIEW

File: components/datetime/src/options/mod.rs [hunk 1]
```diff
@@ -5,69 +5,69 @@
...
+//!     length::Bag {
+//!          date: Some(length::Date::Medium),
+//!          time: Some(length::Time::Short),
 //!         ..Default::default()
 //!     }
 //! );
 //! ```
 //!
-//! At the moment only the `Style` bag works, and we plan to extend that to support
+//! At the moment only the `Length` bag works, and we plan to extend that to support
```
Comment (nciric) [inline components/datetime/src/options/mod.rs:26]: @zbraniecki is this still true or should I remove this future looking comment?

Comment (zbraniecki) [reply components/datetime/src/options/mod.rs:26]: We started adding support for components, but I don't think we can claim ECMA-402 support yet. @gregtatum is driving that.

**Review by zbraniecki — DISMISSED** [review]
lgtm! Thank you!

@Manishearth seems like clippy is overly aggressive due to https://rust-lang.github.io/rust-clippy/master/index.html#upper_case_acronyms - is it possible to make it tone down and respect fully capitalized parts like `CCC8` and `ICU4XFoo` without complaining about switching it to `Ccc8` and `Icu4XFoo`?

Also, we may actually want to switch `MONDAY` to `Monday`.

**Comment by zbraniecki** [comment]
@nciric you have a conflict

**Comment by Manishearth** [comment]
> @Manishearth seems like clippy is overly aggressive due to https://rust-lang.github.io/rust-clippy/master/index.html#upper_case_acronyms - is it possible to make it tone down and respect fully capitalized parts like `CCC8` and `ICU4XFoo` without complaining about switching it to `Ccc8` and `Icu4XFoo`?

You can add exceptions to clippy.toml, but you may also just want to disable the lint

**Review by Manishearth — APPROVED** [review]

**Review by zbraniecki — APPROVED** [review]

**Review by sffc — COMMENTED** [review]
LGTM (sorry for the delay)

**Pull request merged**